index.d.ts

```diff
@@ -0,0 +1,2932 @@
+   * 
+   * Address objects can also be created inline while creating another object, for example during Shipment Creation.
+   * 
+   * Verify an Address
+   *  - Verifying an Address before you ship is a great way to reduce issues with delivery.
+   *    Creating a verified Address is as simple as including an enumerated list of the verifications you'd like EasyPost to perform in the verify or verify_strict url parameters. 
+   *    If any of the verification checks included in the verify_strict list fail an error will be returned from the API. 
+   *    The example below demonstrates the most common verification: "delivery", which checks that the address is deliverable and sets its residential delivery indicator.
+   * 
+   *  - The most effective time to perform address verification is when your customer, or the person entering the delivery address, is present. 
+   *    When designing a shopping cart it is recommended to ask the shopper for their address and verify it on the spot. 
+   *    If verification fails, ask them to double check their input; if they confirm that their data is correct, assume they know their address more correctly than the verification process.
+   * 
+   * @see https://www.easypost.com/docs/api/node#create-and-verify-addresses
+   */
+  public save(): Promise<Address>;
+
+  /**
+   * An Address can be retrieved by its id.
+   * 
+   * @see https://www.easypost.com/docs/api/node#retrieve-an-address
+   * 
+   * @param addressId Unique, begins with "adr_"
+   */
+  static retrieve(addressId: string): Promise<Address>;
+}
+
+
+export declare interface IParcelCreateParameters extends Omit<IParcel, ParametersToOmitOnCreate> { }
+export declare class Parcel implements IParcel {
+  public constructor(input: DeepPartial<IParcelCreateParameters>);
+
+  length: number;
+  width: number;
+  height: number;
+  predefined_package?: PredefinedPackage;
+  weight: number;
+  id: string;
+  mode: "test" | "production";
+  object: "Parcel";
+  created_at: string;
+  updated_at: string;
+
+  /**
+   * Include the weight, and either a predefined_package or length, width and height if applicable.
+   * 
+   * @see https://www.easypost.com/docs/api/node#create-a-parcel
+   */
+  public save(): Promise<Parcel>;
+
+  /**
+   * Get a Parcel by its id. 
+   * In general you should not need to use this in your automated solution. 
+   * A Parcel's id can be inlined into the creation call to other objects. 
+   * This allows you to only create one Parcel for each package you will be using.
+   * 
+   * @see https://www.easypost.com/docs/api/node#retrieve-a-parcel
+   * 
+   * @param parcelId Unique, begins with "prcl_"
+   */
+  static retrieve(parcelId: string): Promise<Parcel>;
+}
+
+export declare interface IShipmentCreateParameters {
+  reference?: string;
+  to_address: IAddress | string;
+  from_address: IAddress | string;
```

Comment:
Should these require partials? Right now it is asking for all fields of an address if you want to create them inline.